### PR TITLE
Ak/main readme

### PR DESCRIPTION
**Pull request by AlexKoff88**
# What does this PR do?

Updated the main README. Added OpenVINO-related documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
...
 
-[OpenVINO](https://docs.openvino.ai/latest/index.html) is an open-source toolkit enabling model optimization and providing high-performance inference solutions for XPUs including various types of CPUs, GPUs, and special DL inference accelerators.
+[OpenVINO](https://docs.openvino.ai/latest/index.html) is an open-source toolkit that enalbes high performance inference capabilities for Intel CPUs, GPUs, and special DL inference accelerators. It also provides a set of tools to optimize and quantize models. Optimum Intel provides a simple interface to convert a model to OpenVINO IR format and to run inference using OpenVINO.
```
Comment (helena-intel) [inline README.md:11]:
```suggestion
[OpenVINO](https://docs.openvino.ai/latest/index.html) is an open-source toolkit that enables high performance inference capabilities for Intel CPUs, GPUs, and special DL inference accelerators. It also provides a set of tools to optimize and quantize models. Optimum Intel provides a simple interface to convert a model to OpenVINO IR format and to run inference using OpenVINO.
```

File: README.md [hunk 2]
```diff
@@ -97,8 +97,9 @@ You can load many more quantized models hosted on the hub under the Intel organi
 Check out the [`examples`](https://github.com/huggingface/optimum-intel/tree/main/examples) directory for more sophisticated usage.
 
 ### OpenVINO
+Below are the examples of how to use OpenVINO and its [NNCF](https://docs.openvino.ai/latest/tmo_introduction.html) tool for model optimization and inference.
```
Comment (helena-intel) [inline README.md:100]:
```suggestion
Below are the examples of how to use OpenVINO and its [NNCF](https://docs.openvino.ai/latest/tmo_introduction.html) framework for model optimization, quantization, and inference.
```
NNCF framework is doubling framework, but to me tool does not cover it.
IMO it would be good to mention quantization specifically, to prevent confusion about the kind of optimization OpenVINO Runtime does and what NNCF does.

File: README.md [hunk 3]
```diff
@@ -112,4 +113,71 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
 pipe_cls = pipeline("text-classification", model=model, tokenizer=tokenizer)
 text = "He's a dreadful magician."
 outputs = pipe_cls(text)
-```
\ No newline at end of file
+```
+
+- Post-training quantizaiton example:
```
Comment (helena-intel) [inline README.md:118]:
```suggestion
- Post-training quantization example:
```

File: README.md [hunk 4]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
++from optimum.intel.openvino import OVModelForSequenceClassification
+
+model_id = "distilbert-base-uncased-finetuned-sst-2-english"
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+dataset = load_dataset("glue", "sst2")
+dataset = dataset.map(
+    lambda examples: tokenizer(examples["sentence"], padding="max_length", max_length=128), batched=True
+)
+compute_metrics = lambda p: metric.compute(
```
Comment (helena-intel) [inline README.md:166]: metric is not defined yet. Easiest is to add `metric = load_metric("accuracy")` but there is a DeprecationWarning that says to use evaluate instead. @echarlaix  should we add evaluate to nncf extra dependencies and use `metric = evaluate.load("accuracy")`?

File: README.md [hunk 5]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+    feature="sequence-classification",
+    args=TrainingArguments('nncf_results', num_train_epochs=1.0, do_train=True, do_eval=True),
+    train_dataset=train_dataset,
+    eval_dataset=eval_dataset,
+    compute_metrics=compute_metrics,
+    tokenizer=tokenizer,
+    data_collator=default_data_collator,
+)
+train_result = trainer.train()
+metrics = trainer.evaluate()
```
Comment (helena-intel) [inline README.md:182]:
```suggestion
metrics = trainer.evaluate()
print(metrics)
```
I know users can add this themselves but if it takes a while to train, IMO it is nice if the copy/pasteable example shows how well it worked.

File: README.md [hunk 6]
```diff
@@ -97,8 +97,9 @@ You can load many more quantized models hosted on the hub under the Intel organi
 Check out the [`examples`](https://github.com/huggingface/optimum-intel/tree/main/examples) directory for more sophisticated usage.
 
 ### OpenVINO
+Below are the examples of how to use OpenVINO and its [NNCF](https://docs.openvino.ai/latest/tmo_introduction.html) framework for model optimization, quantization, and inference.
 
-Here is an example on how to perform inference with OpenVINO Runtime:
+- OpenVINO inference example:
```
Comment (helena-intel) [inline README.md:102]:
```suggestion
#### OpenVINO inference example:
```
IMO it looks nicer if this is a header, and it makes it possible to create a direct link to an example.

File: README.md [hunk 7]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
 pipe_cls = pipeline("text-classification", model=model, tokenizer=tokenizer)
 text = "He's a dreadful magician."
 outputs = pipe_cls(text)
-```
\ No newline at end of file
+```
+
+- Post-training quantizaiton example:
```
Comment (helena-intel) [inline README.md:118]:
```suggestion
#### Post-training quantization example:
```

File: README.md [hunk 8]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+quantizer.quantize(
+    calibration_dataset=calibration_dataset,
+    save_directory='nncf_results' # use this folder as and input for OVModelForSequenceClassification
+                                  # to do the inference
+)
+
+optimized_model = OVModelForSequenceClassification.from_pretrained('nncf_results')
+```
+
+- Quantization-aware training example:
```
Comment (helena-intel) [inline README.md:150]:
```suggestion
#### Quantization-aware training example:
```

File: README.md [hunk 9]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+from transformers import AutoModelForSequenceClassification, AutoTokenizer, TrainingArguments, default_data_collator
+from optimum.intel.openvino.trainer import OVTrainer
++from optimum.intel.openvino import OVModelForSequenceClassification
+
+model_id = "distilbert-base-uncased-finetuned-sst-2-english"
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+dataset = load_dataset("glue", "sst2")
+dataset = dataset.map(
+    lambda examples: tokenizer(examples["sentence"], padding="max_length", max_length=128), batched=True
```
Comment (helena-intel) [inline README.md:164]:
```suggestion
    lambda examples: tokenizer(examples["sentence"], padding=True, truncation=True, max_length=tokenizer.model_max_length), batched=True
```
`padding=True` pads to the largest sequence in the batch, which will usually be shorter than max_length, so this speeds up training/inference time by quite a bit. I added truncation=True for the cases where dataset input is longer than model_max_length. 

tokenizer.model_max_length is not defined for every model so it might be safer to set it to a specific value in this README example (and in the PTQ one), to prevent users running into issues if they modify the example with a different model.

File: README.md [hunk 10]
```diff
@@ -112,4 +113,76 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+from functools import partial
+from optimum.intel.openvino.quantization import OVQuantizer 
+from transformers import AutoTokenizer, AutoModelForSequenceClassification, OVModelForSequenceClassification
+
+model_id = "distilbert-base-uncased-finetuned-sst-2-english"
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+def preprocess_fn(examples, tokenizer):
+    return tokenizer(
+        examples["sentence"], padding="max_length", max_length=tokenizer.model_max_length, truncation=True
```
Comment (helena-intel) [inline README.md:129]:
```suggestion
        examples["sentence"], padding=True, truncation=True, max_length=tokenizer.model_max_length
```
See comment below for QAT example

File: README.md (hunk 9 above)
Comment (AlexKoff88) [reply README.md:164]: I agree and it would be great to have it as generic as possible.

File: README.md [hunk 11]
```diff
@@ -112,4 +113,71 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+compute_metrics = lambda p: metric.compute(
+    predictions=np.argmax(p.predictions, axis=1), references=p.label_ids
+)
+
+-trainer = Trainer(
++trainer = OVTrainer(
+    model=model,
+    feature="sequence-classification",
+    args=TrainingArguments('nncf_results', num_train_epochs=1.0, do_train=True, do_eval=True),
+    train_dataset=train_dataset,
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply README.md:175]: I like the suggestion. It would be helpful for those who does not have GPU but would like to see how the tuning works in practice.

**Comment by AlexKoff88** [comment]
@kding1, please take a look and say if you are ok with the structure.

File: README.md [hunk 12]
```diff
@@ -27,8 +27,8 @@ pip install git+https://github.com/huggingface/optimum-intel.git
 To install the latest release of this package with the corresponding required dependencies, you can do respectively:
 | Accelerator                                                                                                      | Installation                                      |
 |:-----------------------------------------------------------------------------------------------------------------|:--------------------------------------------------|
-| [OpenVINO](https://docs.openvino.ai/latest/index.html)                                                           | `python -m pip install optimum[openvino]`         |
 | [Intel Neural Compressor](https://www.intel.com/content/www/us/en/developer/tools/oneapi/neural-compressor.html) | `python -m pip install optimum[neural-compressor]`|
+| [OpenVINO](https://docs.openvino.ai/latest/index.html)                                                           | `python -m pip install optimum[openvino,nncf]`         |
```
[the remark this answers is not in the document]
Comment (echarlaix) [reply README.md:31]: Yes you are right ! We will be able to change it back after optimum next release

File: README.md [hunk 13]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+    dataset_split="train",
+    preprocess_batch=True,
+)
+quantizer.quantize(
+    calibration_dataset=calibration_dataset,
+    save_directory='nncf_results' # use this folder as and input for OVModelForSequenceClassification
+                                  # to do the inference
+)
+
+optimized_model = OVModelForSequenceClassification.from_pretrained('nncf_results')
```
Comment (echarlaix) [inline README.md:147]:
```suggestion
# The directory where the quantized model will be saved
save_dir = "nncf_results"
# Apply static quantization and save the resulting model in the OpenVINO IR format
quantizer.quantize(calibration_dataset=calibration_dataset, save_directory=save_dir)
# Load the quantized model
optimized_model = OVModelForSequenceClassification.from_pretrained(save_dir)
```

File: README.md [hunk 14]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+dataset = load_dataset("glue", "sst2")
+dataset = dataset.map(
+    lambda examples: tokenizer(examples["sentence"], padding=True, truncation=True, max_length=tokenizer.model_max_length), batched=True
+)
+compute_metrics = lambda p: metric.compute(
+    predictions=np.argmax(p.predictions, axis=1), references=p.label_ids
+)
+
```
Comment (echarlaix) [inline README.md:169]:
```
# The directory where the quantized model will be saved
save_dir = "nncf_results"
```

File: README.md [hunk 15]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+-trainer = Trainer(
++trainer = OVTrainer(
+    model=model,
+    feature="sequence-classification",
+    args=TrainingArguments('nncf_results', num_train_epochs=1.0, do_train=True, do_eval=True),
```
Comment (echarlaix) [inline README.md:174]:
```suggestion
    args=TrainingArguments(save_dir, num_train_epochs=1.0, do_train=True, do_eval=True),
```

File: README.md [hunk 16]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+    compute_metrics=compute_metrics,
+    tokenizer=tokenizer,
+    data_collator=default_data_collator,
+)
+train_result = trainer.train()
+metrics = trainer.evaluate()
+print(metrics)
+trainer.save_model()
+
++optimized_model = OVModelForSequenceClassification.from_pretrained('nncf_results')
```
Comment (echarlaix) [inline README.md:186]:
```suggestion
+optimized_model = OVModelForSequenceClassification.from_pretrained(save_dir)
```

File: README.md [hunk 17]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+```
+
+You can find more OpenVINO examples in the corresponding Optimum-Intel documentation.
```
Comment (echarlaix) [inline README.md:189]:
```suggestion
You can find more OpenVINO examples in the corresponding Optimum Intel documentation.
```

File: README.md [hunk 18]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+from optimum.intel.openvino.trainer import OVTrainer
++from optimum.intel.openvino import OVModelForSequenceClassification
+
+model_id = "distilbert-base-uncased-finetuned-sst-2-english"
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+dataset = load_dataset("glue", "sst2")
+dataset = dataset.map(
+    lambda examples: tokenizer(examples["sentence"], padding=True, truncation=True, max_length=tokenizer.model_max_length), batched=True
+)
```
Comment (echarlaix) [inline README.md:165]:
```suggestion
dataset = dataset.map(
    lambda examples: tokenizer(examples["sentence"], padding=True, truncation=True, max_length=128), batched=True
)
```

File: README.md [hunk 19]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+optimized_model = OVModelForSequenceClassification.from_pretrained('nncf_results')
+```
+
+#### Quantization-aware training example:
+
+```diff
+import numpy as np
+from datasets import load_dataset, load_metric
+from transformers import AutoModelForSequenceClassification, AutoTokenizer, TrainingArguments, default_data_collator
+from optimum.intel.openvino.trainer import OVTrainer
```
Comment (echarlaix) [inline README.md:156]:
```suggestion
-from transformers import Trainer
+from optimum.intel.openvino.trainer import OVTrainer
```

File: README.md [hunk 20]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+    args=TrainingArguments('nncf_results', num_train_epochs=1.0, do_train=True, do_eval=True),
+    train_dataset=dataset["train"].select(range(300)),
+    eval_dataset=dataset["validation"],
+    compute_metrics=compute_metrics,
+    tokenizer=tokenizer,
+    data_collator=default_data_collator,
+)
+train_result = trainer.train()
+metrics = trainer.evaluate()
+print(metrics)
```
Comment (echarlaix) [inline README.md:183]: I would remove this print

File: README.md [hunk 21]
```diff
@@ -112,4 +113,77 @@ tokenizer = AutoTokenizer.from_pretrained(model_id)
...
+from functools import partial
+from optimum.intel.openvino.quantization import OVQuantizer 
+from transformers import AutoTokenizer, AutoModelForSequenceClassification, OVModelForSequenceClassification
+
+model_id = "distilbert-base-uncased-finetuned-sst-2-english"
+model = AutoModelForSequenceClassification.from_pretrained(model_id)    
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+def preprocess_fn(examples, tokenizer):
+    return tokenizer(
+        examples["sentence"], padding=True, truncation=True, max_length=tokenizer.model_max_length
```
Comment (echarlaix) [inline README.md:129]:
```suggestion
        examples["sentence"], padding=True, truncation=True, max_length=128
```

File: README.md (hunk 4 above)
Comment (echarlaix) [reply README.md:166]: Yes it's a great point

File: README.md (hunk 20 above)
Comment (AlexKoff88) [reply README.md:183]: @helena-intel, asked to add this. I assume it helps to improve UX since we expect that this will be copy-pasted

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply README.md]: @echarlaix, can we say that this is resolved?

**Review by echarlaix — APPROVED** [review]
Thanks for this addition @AlexKoff88 !

File: docs/source/index.mdx [hunk 22]
```diff
@@ -15,10 +15,30 @@ limitations under the License.
...
 
 [Intel Neural Compressor](https://www.intel.com/content/www/us/en/developer/tools/oneapi/neural-compressor.html) is an open-source library enabling the usage of the most popular compression techniques such as quantization, pruning and knowledge distillation. It supports automatic accuracy-driven tuning strategies in order for users to easily generate quantized model. The users can easily apply static, dynamic and aware-training quantization approaches while giving an expected accuracy criteria. It also supports different weight pruning techniques enabling the creation of pruned model giving a predefined sparsity target.
 
-[OpenVINO](https://docs.openvino.ai/latest/index.html) is an open-source toolkit enabling model optimization and providing high-performance inference solutions for XPUs including various types of CPUs, GPUs, and special DL inference accelerators.
\ No newline at end of file
+[OpenVINO](https://docs.openvino.ai/latest/index.html) is an open-source toolkit enabling model optimization and providing high-performance inference solutions for XPUs including various types of CPUs, GPUs, and special DL inference accelerators.
```
Comment (AlexKoff88) [inline docs/source/index.mdx:24]: Can we have the text about OpenVINO from the main README here as well?
